### ISSUE TITLE
issue related to pillow?

### ISSUE THREAD

**Issue by kevinrue** [kevinrue/velociraptor#78]
https://github.com/python-pillow/Pillow/issues/8148

**Comment by kevinrue** [comment]
Error:

```
Quitting from lines 76-82 [unnamed-chunk-5] (velociraptor.Rmd)
Error: processing vignette 'velociraptor.Rmd' failed with diagnostics:
ImportError: /github/home/.cache/R/basilisk/1.17.2/velociraptor/1.15.4/env/lib/python3.10/site-packages/PIL/../../../libtiff.so.6: undefined symbol: jpeg12_write_raw_data, version LIBJPEG_8.0
Run `reticulate::py_last_error()` for details.
--- failed re-building ‘velociraptor.Rmd’
```

Also for the record, `mamba search scvelo -c conda-forge -c bioconda`:

```
# Name                       Version           Build  Channel             
scvelo                        0.1.24            py_0  bioconda            
scvelo                        0.1.25            py_0  bioconda            
scvelo                         0.2.1            py_0  bioconda            
scvelo                         0.2.2            py_0  bioconda            
scvelo                         0.2.2            py_1  bioconda            
scvelo                         0.2.3            py_0  bioconda            
scvelo                         0.2.4    pyhdfd78af_0  bioconda            
scvelo                         0.2.5    pyhd8ed1ab_0  conda-forge         
scvelo                         0.2.5    pyhdfd78af_0  bioconda            
scvelo                         0.3.0    pyhd8ed1ab_0  conda-forge         
scvelo                         0.3.1    pyhd8ed1ab_0  conda-forge         
scvelo                         0.3.1    pyhd8ed1ab_1  conda-forge         
scvelo                         0.3.2    pyhd8ed1ab_1  conda-forge 
```

**Comment by kevinrue** [comment]
installed scvelo 0.3.2 manually using micromamba

```
micromamba create -n scvelo -c conda-forge scvelo
```

ran into https://github.com/theislab/scvelo/issues/1260

Next attempt:

```
micromamba create -n scvelo -c conda-forge scvelo==0.3.2 scipy==1.13.1
```

**Comment by kevinrue** [comment]
Micromamba environment and Python script that simulate the velociraptor workflow:

```
name: scvelo
channels:
- conda-forge
dependencies:
- _libgcc_mutex=0.1=conda_forge
- _openmp_mutex=4.5=2_gnu
- absl-py=2.1.0=pyhd8ed1ab_0
- anndata=0.10.8=pyhd8ed1ab_0
- annotated-types=0.7.0=pyhd8ed1ab_0
- anyio=4.4.0=pyhd8ed1ab_0
- aom=3.9.1=hac33072_0
- array-api-compat=1.8=pyhd8ed1ab_0
- arrow=1.3.0=pyhd8ed1ab_0
- backoff=2.2.1=pyhd8ed1ab_0
- beautifulsoup4=4.12.3=pyha770c72_0
- blessed=1.19.1=pyhe4f9e05_2
- blosc=1.21.6=hef167b5_0
- boto3=1.35.3=pyhd8ed1ab_0
- botocore=1.35.3=pyge310_1234567_0
- brotli=1.1.0=hd590300_1
- brotli-bin=1.1.0=hd590300_1
- brotli-python=1.1.0=py311hb755f60_1
- bzip2=1.0.8=h4bc722e_7
- c-ares=1.33.0=ha66036c_0
- ca-certificates=2024.7.4=hbcca054_0
- cachecontrol=0.14.0=pyhd8ed1ab_1
- cachecontrol-with-filecache=0.14.0=pyhd8ed1ab_1
- cached-property=1.5.2=hd8ed1ab_1
- cached_property=1.5.2=pyha770c72_1
- certifi=2024.7.4=pyhd8ed1ab_0
- cffi=1.17.0=py311ha8e6434_0
- charset-normalizer=3.3.2=pyhd8ed1ab_0
- chex=0.1.86=pyhd8ed1ab_0
- cleo=2.1.0=pyhd8ed1ab_0
- click=8.1.7=unix_pyh707e725_0
- colorama=0.4.6=pyhd8ed1ab_0
- contextlib2=21.6.0=pyhd8ed1ab_0
- contourpy=1.2.1=py311h9547e67_0
- crashtest=0.4.1=pyhd8ed1ab_0
- croniter=1.4.1=pyhd8ed1ab_0
- cryptography=43.0.0=py311hc6616f6_0
- cycler=0.12.1=pyhd8ed1ab_0
- dateutils=0.6.12=py_0
- dav1d=1.2.1=hd590300_0
- dbus=1.13.6=h5008d03_3
- deepdiff=7.0.1=pyhd8ed1ab_0
- distlib=0.3.8=pyhd8ed1ab_0
- dnspython=2.6.1=pyhd8ed1ab_1
- docrep=0.3.2=pyh44b312d_0
- dulwich=0.21.7=py311h459d7ec_0
- email-validator=2.2.0=pyhd8ed1ab_0
- email_validator=2.2.0=hd8ed1ab_0
- et_xmlfile=1.1.0=pyhd8ed1ab_0
- etils=1.9.2=pyhd8ed1ab_0
- exceptiongroup=1.2.2=pyhd8ed1ab_0
- expat=2.6.2=h59595ed_0
- fastapi=0.112.1=pyhd8ed1ab_0
- fastapi-cli=0.0.5=pyhd8ed1ab_0
- filelock=3.15.4=pyhd8ed1ab_0
- flax=0.8.5=pyhd8ed1ab_0
- fonttools=4.53.1=py311h61187de_0
- freetype=2.12.1=h267a509_2
- fsspec=2024.6.1=pyhff2d567_0
- get-annotations=0.1.2=pyhd8ed1ab_0
- gmp=6.3.0=hac33072_2
- gmpy2=2.1.5=py311hc4f1f91_1
- h11=0.14.0=pyhd8ed1ab_0
- h2=4.1.0=pyhd8ed1ab_0
- h5py=3.11.0=nompi_py311h439e445_102
- hdf5=1.14.3=nompi_hdf9ad27_105
- hpack=4.0.0=pyh9f0ad1d_0
- httpcore=1.0.5=pyhd8ed1ab_0
- httpx=0.27.0=pyhd8ed1ab_0
- hyperframe=6.0.1=pyhd8ed1ab_0
- icu=75.1=he02047a_0
- idna=3.7=pyhd8ed1ab_0
- importlib-metadata=8.4.0=pyha770c72_0
- importlib_metadata=8.4.0=hd8ed1ab_0
- importlib_resources=6.4.4=pyhd8ed1ab_0
- inquirer=3.1.4=pyhd8ed1ab_0
- itsdangerous=2.2.0=pyhd8ed1ab_0
- jaraco.classes=3.4.0=pyhd8ed1ab_1
- jax=0.4.31=pyhd8ed1ab_0
- jaxlib=0.4.31=cpu_py311hb2c720c_0
- jeepney=0.8.0=pyhd8ed1ab_0
- jinja2=3.1.4=pyhd8ed1ab_0
- jmespath=1.0.1=pyhd8ed1ab_0
- joblib=1.4.2=pyhd8ed1ab_0
- keyring=24.3.1=py311h38be061_0
- keyutils=1.6.1=h166bdaf_0
- kiwisolver=1.4.5=py311h9547e67_1
- krb5=1.21.3=h659f571_0
- lcms2=2.16=hb7c19ff_0
- ld_impl_linux-64=2.40=hf3520f5_7
- legacy-api-wrap=1.4=pyhd8ed1ab_1
- lerc=4.0.0=h27087fc_0
- libabseil=20240116.2=cxx17_he02047a_1
- libaec=1.1.3=h59595ed_0
- libavif16=1.1.1=h9b56c87_0
- libblas=3.9.0=23_linux64_openblas
- libbrotlicommon=1.1.0=hd590300_1
- libbrotlidec=1.1.0=hd590300_1
- libbrotlienc=1.1.0=hd590300_1
- libcblas=3.9.0=23_linux64_openblas
- libcurl=8.9.1=hdb1bdb2_0
- libdeflate=1.21=h4bc722e_0
- libedit=3.1.20191231=he28a2e2_2
- libev=4.33=hd590300_2
- libexpat=2.6.2=h59595ed_0
- libffi=3.4.2=h7f98852_5
- libgcc-ng=14.1.0=h77fa898_0
- libgfortran-ng=14.1.0=h69a702a_0
- libgfortran5=14.1.0=hc5f4f2c_0
- libglib=2.80.3=h315aac3_2
- libgomp=14.1.0=h77fa898_0
- libgrpc=1.62.2=h15f2491_0
- libhwloc=2.11.1=default_hecaa2ac_1000
- libiconv=1.17=hd590300_2
- libjpeg-turbo=3.0.0=hd590300_1
- liblapack=3.9.0=23_linux64_openblas
- libllvm14=14.0.6=hcd5def8_4
- libnghttp2=1.58.0=h47da74e_1
- libnsl=2.0.1=hd590300_0
- libopenblas=0.3.27=pthreads_hac2b453_1
- libpng=1.6.43=h2797004_0
- libprotobuf=4.25.3=h08a7969_0
- libre2-11=2023.09.01=h5a48ba9_2
- libsqlite=3.46.0=hde9e2c9_0
- libssh2=1.11.0=h0841786_0
- libstdcxx-ng=14.1.0=hc0a3c3a_0
- libtiff=4.6.0=h46a8edc_4
- libtorch=2.3.1=cpu_generic_h970db74_0
- libuuid=2.38.1=h0b41bf4_0
- libuv=1.48.0=hd590300_0
- libwebp-base=1.4.0=hd590300_0
- libxcb=1.16=hd590300_0
- libxcrypt=4.4.36=hd590300_1
- libxml2=2.12.7=he7c6b58_4
- libzlib=1.3.1=h4ab18f5_1
- lightning=2.0.9.post0=pyhd8ed1ab_0
- lightning-cloud=0.5.70=pyhd8ed1ab_0
- lightning-utilities=0.11.6=pyhd8ed1ab_0
- llvmlite=0.43.0=py311hbde99c3_0
- loompy=3.0.6=py_0
- lz4-c=1.9.4=hcb278e6_0
- markdown-it-py=3.0.0=pyhd8ed1ab_0
- markupsafe=2.1.5=py311h459d7ec_0
- matplotlib-base=3.9.2=py311h74b4f7c_0
- mdurl=0.1.2=pyhd8ed1ab_0
- ml-collections=0.1.1=pyhd8ed1ab_0
- ml_dtypes=0.4.0=py311h14de704_1
- more-itertools=10.4.0=pyhd8ed1ab_0
- mpc=1.3.1=hfe3b2da_0
- mpfr=4.2.1=h38ae2d0_2
- mpmath=1.3.0=pyhd8ed1ab_0
- msgpack-python=1.0.8=py311h52f7536_0
- mudata=0.3.0=pyhd8ed1ab_0
- multipledispatch=0.6.0=pyhd8ed1ab_1
- munkres=1.1.4=pyh9f0ad1d_0
- natsort=8.4.0=pyhd8ed1ab_0
- ncurses=6.5=h59595ed_0
- nest-asyncio=1.6.0=pyhd8ed1ab_0
- networkx=3.3=pyhd8ed1ab_1
- nomkl=1.0=h5ca1d4c_0
- numba=0.60.0=py311h4bc866e_0
- numpy=1.26.4=py311h64a7726_0
- numpy_groupies=0.11.2=pyhd8ed1ab_0
- numpyro=0.15.2=pyhd8ed1ab_0
- openjpeg=2.5.2=h488ebb8_0
- openpyxl=3.1.5=py311h459d7ec_0
- openssl=3.3.1=h4bc722e_2
- opt-einsum=3.3.0=hd8ed1ab_2
- opt_einsum=3.3.0=pyhc1e730c_2
- optax=0.2.2=pyhd8ed1ab_1
- orbax-checkpoint=0.4.4=pyhd8ed1ab_0
- ordered-set=4.1.0=pyhd8ed1ab_0
- orjson=3.10.7=py311hb3a8bbb_0
- packaging=24.1=pyhd8ed1ab_0
- pandas=2.2.2=py311h14de704_1
- patsy=0.5.6=pyhd8ed1ab_0
- pcre2=10.44=hba22ea6_2
- pexpect=4.9.0=pyhd8ed1ab_0
- pillow=10.4.0=py311h82a398c_0
- pip=24.2=pyhd8ed1ab_0
- pkginfo=1.11.1=pyhd8ed1ab_0
- platformdirs=4.2.2=pyhd8ed1ab_0
- poetry=1.8.3=linux_pyha804496_1
- poetry-core=1.9.0=pyhd8ed1ab_0
- poetry-plugin-export=1.8.0=pyhd8ed1ab_0
- protobuf=4.25.3=py311h7b78aeb_0
- psutil=6.0.0=py311h331c9d8_0
- pthread-stubs=0.4=h36c2ea0_1001
- ptyprocess=0.7.0=pyhd3deb0d_0
- pybind11-abi=4=hd8ed1ab_3
- pycparser=2.22=pyhd8ed1ab_0
- pydantic=2.1.1=pyhd8ed1ab_0
- pydantic-core=2.4.0=py311h46250e7_0
- pygments=2.18.0=pyhd8ed1ab_0
- pyjwt=2.9.0=pyhd8ed1ab_1
- pynndescent=0.5.13=pyhff2d567_0
- pyparsing=3.1.2=pyhd8ed1ab_0
- pyproject_hooks=1.1.0=pyhd8ed1ab_0
- pyro-api=0.1.2=pyhd8ed1ab_0
- pyro-ppl=1.9.1=pyhd8ed1ab_0
- pysocks=1.7.1=pyha2e5f31_6
- python=3.11.9=hb806964_0_cpython
- python-build=1.2.1=pyhd8ed1ab_0
- python-dateutil=2.9.0=pyhd8ed1ab_0
- python-editor=1.0.4=pyhd8ed1ab_1
- python-fastjsonschema=2.20.0=pyhd8ed1ab_0
- python-installer=0.7.0=pyhd8ed1ab_0
- python-multipart=0.0.9=pyhd8ed1ab_0
- python-tzdata=2024.1=pyhd8ed1ab_0
- python_abi=3.11=5_cp311
- pytorch=2.3.1=cpu_generic_py311h8ca351a_0
- pytorch-lightning=2.4.0=pyhd8ed1ab_0
- pytz=2024.1=pyhd8ed1ab_0
- pyyaml=6.0.2=py311h61187de_0
- qhull=2020.2=h434a139_5
- rapidfuzz=3.9.6=py311hf86e51f_0
- rav1e=0.6.6=he8a937b_2
- re2=2023.09.01=h7f4b329_2
- readchar=4.2.0=pyhd8ed1ab_0
- readline=8.2=h8228510_1
- requests=2.32.3=pyhd8ed1ab_0
- requests-toolbelt=1.0.0=pyhd8ed1ab_0
- rich=13.7.1=pyhd8ed1ab_0
- s3transfer=0.10.2=pyhd8ed1ab_0
- scanpy=1.10.2=pyhd8ed1ab_0
- scikit-learn=1.1.3=py311h3b52e38_1
- scipy=1.13.1=py311h517d4fd_0
- scvelo=0.3.2=pyhd8ed1ab_1
- scvi-tools=1.1.6=pyhd8ed1ab_0
- seaborn=0.13.2=hd8ed1ab_2
- seaborn-base=0.13.2=pyhd8ed1ab_2
- secretstorage=3.3.3=py311h38be061_2
- session-info=1.0.0=pyhd8ed1ab_0
- setuptools=72.2.0=pyhd8ed1ab_0
- shellingham=1.5.4=pyhd8ed1ab_0
- six=1.16.0=pyh6c4a22f_0
- sleef=3.6.1=h3400bea_1
- snappy=1.2.1=ha2e4443_0
- sniffio=1.3.1=pyhd8ed1ab_0
- soupsieve=2.5=pyhd8ed1ab_1
- sparse=0.15.4=pyhd8ed1ab_0
- starlette=0.38.2=pyhd8ed1ab_0
- starsessions=1.3.0=pyhd8ed1ab_0
- statsmodels=0.14.2=py311h18e1886_0
- stdlib-list=0.10.0=pyhd8ed1ab_0
- svt-av1=2.1.2=hac33072_0
- sympy=1.13.2=pypyh2585a3b_103
- tbb=2021.12.0=h434a139_3
- tensorstore=0.1.62=py311h865b1b2_0
- threadpoolctl=3.5.0=pyhc1e730c_0
- tk=8.6.13=noxft_h4845f30_101
- tomli=2.0.1=pyhd8ed1ab_0
- tomlkit=0.13.2=pyha770c72_0
- toolz=0.12.1=pyhd8ed1ab_0
- torchmetrics=1.4.0.post0=pyhd8ed1ab_0
- tqdm=4.66.5=pyhd8ed1ab_0
- traitlets=5.14.3=pyhd8ed1ab_0
- trove-classifiers=2024.7.2=pyhd8ed1ab_0
- typer=0.12.4=pyhd8ed1ab_0
- typer-slim=0.12.4=pyhd8ed1ab_0
- typer-slim-standard=0.12.4=hd8ed1ab_0
- types-python-dateutil=2.9.0.20240821=pyhd8ed1ab_0
- typing-extensions=4.12.2=hd8ed1ab_0
- typing_extensions=4.12.2=pyha770c72_0
- tzdata=2024a=h0c530f3_0
- umap-learn=0.5.6=py311h38be061_1
- urllib3=2.2.2=pyhd8ed1ab_1
- uvicorn=0.30.6=py311h38be061_0
- virtualenv=20.26.3=pyhd8ed1ab_0
- wcwidth=0.2.13=pyhd8ed1ab_0
- websocket-client=1.8.0=pyhd8ed1ab_0
- websockets=12.0=py311h459d7ec_0
- wheel=0.44.0=pyhd8ed1ab_0
- xarray=2024.7.0=pyhd8ed1ab_0
- xorg-libxau=1.0.11=hd590300_0
- xorg-libxdmcp=1.1.3=h7f98852_0
- xz=5.2.6=h166bdaf_0
- yaml=0.2.5=h7f98852_2
- zipp=3.20.0=pyhd8ed1ab_0
- zstandard=0.23.0=py311h5cd10c7_0
- zstd=1.5.6=ha6fb4c9_0
```

Python script
```
import scvelo as scv
import scanpy as sc
scv.logging.print_version()
#
scv.settings.verbosity = 3  # show errors(0), warnings(1), info(2), hints(3)
scv.settings.presenter_view = True  # set max width size for presenter view
scv.set_figure_params('scvelo')  # for beautified visualization
#
adata = scv.datasets.pancreas()
adata
#
scv.pl.proportions(adata)
#
scv.pp.filter_genes(adata, min_shared_counts=20)
scv.pp.normalize_per_cell(adata)
scv.pp.filter_genes_dispersion(adata, n_top_genes=2000)
scv.pp.log1p(adata)
#
scv.pp.filter_and_normalize(adata, min_shared_counts=20, n_top_genes=2000)
sc.pp.neighbors(adata)
scv.pp.moments(adata, n_pcs=None, n_neighbors=None)
#
scv.tl.velocity(adata)
```

**Comment by kevinrue** [comment]
Summary:

> the error only happens in the context of R CMD build, but everything works fine interactively or in the context of R CMD check

https://community-bioc.slack.com/archives/CM2CUGBGB/p1717516918056459?thread_ts=1713725998.833219&cid=CM2CUGBGB

**Comment by kevinrue** [comment]
What about this https://bugs.archlinux.org/task/79579.html ?

**Comment by kevinrue** [comment]
pinning Vince's comment; https://community-bioc.slack.com/archives/CM2CUGBGB/p1717520618885429?thread_ts=1713725998.833219&cid=CM2CUGBGB

> the different phases of R build vs check could use, unintentionally, different dynamic libs for python modules

**Comment by kevinrue** [comment]
Fixed by #81
